controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -63,20 +65,19 @@ func NewAdapter(pipelineRun *tektonv1.PipelineRun, application *applicationapiv1
 // to be consumed by user
 func (a *Adapter) EnsureStatusReportedInSnapshot() (controller.OperationResult, error) {
 	var pipelinerunStatus intgteststat.IntegrationTestStatus
-	var snapshot *applicationapiv1alpha1.Snapshot
 	var detail string
 	var err error
 
 	// pipelines run in parallel and have great potential to cause conflict on update
 	// thus `RetryOnConflict` is easy solution here, given the snapshot must be loaded specifically here
 	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
 
-		snapshot, err = a.loader.GetSnapshotFromPipelineRun(a.client, a.context, a.pipelineRun)
+		a.snapshot, err = a.loader.GetSnapshotFromPipelineRun(a.client, a.context, a.pipelineRun)
```

Comment:
I feel this can be removed since it is called already [here](https://github.com/redhat-appstudio/integration-service/pull/427/files#diff-4988e4b24de307c0d7aebf7a96fdc2dd512ca67f53c303efe0c3c431e9e0e8bdR90).

Reply:
That was my initial idea too, but this block retries on conflict when trying to update the Snapshot, so the Snapshot still needs to be re-loaded from the cluster in case the resource changed in the meantime.

Reply:
Thanks.